### PR TITLE
Implemented issue #59 containsNotKey for Map

### PR DESCRIPTION
**Pull request by uaArsen**
Issue: #59 
Note: I renamed containsNotKey to notContainsKey 
Added assertion notContainsKey to modules:
 * api-cc-en_GB-common
 * api-infix-cc-en_GB-common
 * api-cc-de_CH-common -- renamed it enthaeltNichtKey

----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/cc-de_CH/atrium-api-cc-de_CH-common/src/main/kotlin/ch/tutteli/atrium/api/cc/de_CH/mapAssertions.kt [hunk 1]
```diff
@@ -81,6 +81,15 @@ inline fun <K, reified V : Any, T: Map<K, V?>> Assert<T>.enthaeltNullable(keyVal
 fun <K> Assert<Map<K, *>>.enthaeltKey(key: K)
     = addAssertion(AssertImpl.map.containsKey(this, key))
 
+/**
+ * Makes the assertion that [AssertionPlant.subject] contains the given [key].
```
Comment (robstoll) [inline apis/cc-de_CH/atrium-api-cc-de_CH-common/src/main/kotlin/ch/tutteli/atrium/api/cc/de_CH/mapAssertions.kt]: Makes the assertion that [AssertionPlant.subject] does not contain the given [key].

File: apis/cc-de_CH/atrium-api-cc-de_CH-common/src/main/kotlin/ch/tutteli/atrium/api/cc/de_CH/mapAssertions.kt [hunk 2]
```diff
@@ -81,6 +81,15 @@ inline fun <K, reified V : Any, T: Map<K, V?>> Assert<T>.enthaeltNullable(keyVal
...
+ *
+ * @return This plant to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ */
+fun <K> Assert<Map<K, *>>.nichtEnthaeltKey(key: K)
```
Comment (robstoll) [inline apis/cc-de_CH/atrium-api-cc-de_CH-common/src/main/kotlin/ch/tutteli/atrium/api/cc/de_CH/mapAssertions.kt]: rename to `enthaeltNichtKey`

File: apis/cc-en_GB/atrium-api-cc-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/en_GB/mapAssertions.kt [hunk 3]
```diff
@@ -80,6 +80,15 @@ inline fun <K, reified V : Any, T: Map<K, V?>> Assert<T>.containsNullable(keyVal
 fun <K> Assert<Map<K, *>>.containsKey(key: K)
     = addAssertion(AssertImpl.map.containsKey(this, key))
 
+/**
+ * Makes the assertion that [AssertionPlant.subject] not contains the given [key].
```
Comment (robstoll) [inline apis/cc-en_GB/atrium-api-cc-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/en_GB/mapAssertions.kt]: Makes the assertion that [AssertionPlant.subject] does not contain the given [key].

File: apis/cc-en_GB/atrium-api-cc-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/en_GB/mapAssertions.kt [hunk 4]
```diff
@@ -80,6 +80,15 @@ inline fun <K, reified V : Any, T: Map<K, V?>> Assert<T>.containsNullable(keyVal
...
+ *
+ * @return This plant to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ */
+fun <K> Assert<Map<K, *>>.notContainsKey(key: K)
```
Comment (robstoll) [inline apis/cc-en_GB/atrium-api-cc-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/cc/en_GB/mapAssertions.kt]: containsNotKey

File: misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt [hunk 5]
```diff
@@ -295,6 +307,30 @@ abstract class MapAssertionsSpec(
         fluent.containsKeyFun("a")
     }
 
+    describeFun(notContainsKey) {
+        it("does not throw if the map not contains the key") {
```
Comment (robstoll) [inline misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt]: does not contain

File: misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt [hunk 6]
```diff
@@ -295,6 +307,30 @@ abstract class MapAssertionsSpec(
...
+            fluent.notContainsKeyFun("c")
+        }
+
+        it("throws an AssertionError if the map does contain the key") {
```
Comment (robstoll) [inline misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt]: if the map contains the key

File: translations/de_CH/atrium-translations-de_CH-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionMapAssertion.kt [hunk 7]
```diff
@@ -10,6 +10,7 @@ enum class DescriptionMapAssertion(override val value: String) : StringBasedTran
     CANNOT_EVALUATE_KEY_DOES_NOT_EXIST("$COULD_NOT_EVALUATE_DEFINED_ASSERTIONS -- der gegebene Key existiert nicht.\n$VISIT_COULD_NOT_EVALUATE_ASSERTIONS"),
     CONTAINS_IN_ANY_ORDER("enthält, in beliebiger Reihenfolge"),
     CONTAINS_KEY("enthält Key"),
+    NOT_CONTAINS_KEY("enthält nicht den Key"),
```
Comment (robstoll) [inline translations/de_CH/atrium-translations-de_CH-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionMapAssertion.kt]: CONTAINS_NOT_KEY

File: misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt [hunk 8]
```diff
@@ -91,6 +102,7 @@ abstract class MapAssertionsSpec(
     val isNotDescr = DescriptionBasic.IS_NOT.getDefault()
     val empty = DescriptionCollectionAssertion.EMPTY.getDefault()
     val containsKeyDescr = DescriptionMapAssertion.CONTAINS_KEY.getDefault()
+    val notContainsKeyDescr = DescriptionMapAssertion.CONTAINS_KEY.getDefault()
```
Comment (robstoll) [inline misc/atrium-spec/src/main/kotlin/ch/tutteli/atrium/spec/integration/MapAssertionsSpec.kt]: CONTAINS_NOT_KEY

**Comment by uaArsen** [comment]
I cant join a Slack channel that you mentioned. It looks like this is: kotlinlang.slack.com .
image

**Comment by uaArsen** [comment]
Fixed according to review. Thank you @robstoll

**Comment by robstoll** [comment]
You can invite yourself to the slack channel via this link: http://slack.kotlinlang.org/

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/MapAssertionsSpec.kt [hunk 9]
```diff
@@ -62,6 +64,12 @@ class MapAssertionsSpec : ch.tutteli.atrium.spec.integration.MapAssertionsSpec(
         private fun containsNullableKey(plant: Assert<Map<String?, *>>, key: String?)
             = plant containsKey key
 
+        private fun notContainsKey(plant: Assert<Map<String, *>>, key: String)
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/MapAssertionsSpec.kt]: rename as well to be consistent

File: apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/MapAssertionsSpec.kt [hunk 10]
```diff
@@ -62,6 +64,12 @@ class MapAssertionsSpec : ch.tutteli.atrium.spec.integration.MapAssertionsSpec(
...
+            = plant containsNotKey key
+
+        private fun notContainsNullableKey(plant: Assert<Map<String?, *>>, key: String?)
```
Comment (robstoll) [inline apis/cc-infix-en_GB/atrium-api-cc-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/cc/infix/en_GB/MapAssertionsSpec.kt]: rename

File: translations/en_GB/atrium-translations-en_GB-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionMapAssertion.kt [hunk 11]
```diff
@@ -10,6 +10,7 @@ enum class DescriptionMapAssertion(override val value: String) : StringBasedTran
     CANNOT_EVALUATE_KEY_DOES_NOT_EXIST("$COULD_NOT_EVALUATE_DEFINED_ASSERTIONS -- given key does not exist.\n$VISIT_COULD_NOT_EVALUATE_ASSERTIONS"),
     CONTAINS_IN_ANY_ORDER("contains, in any order"),
     CONTAINS_KEY("contains key"),
+    CONTAINS_NOT_KEY("not contains key"),
```
Comment (robstoll) [inline translations/en_GB/atrium-translations-en_GB-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionMapAssertion.kt]: does not contain key

**Comment by robstoll** [comment]
You still have WIP in the title, do you intend to do more work? Otherwise I would merge your PR

**Comment by uaArsen** [comment]
I think you can merge PR

**Pull request merged**